<!-- Demonstation of obstacle detector -->
<launch>

  <param name="use_sim_time" value="true"/>  

  <node name="nodelet_manager" pkg="nodelet" type="nodelet" args="manager" output="screen">
    <param name="num_worker_threads" value="20"/>
  </node>

  <node name="obstacle_extractor" pkg="nodelet" type="nodelet" args="load obstacle_detector/ObstacleExtractor nodelet_manager">
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="false"/>
    <param name="circles_from_visibles"  value="false"/>
    <param name="discard_converted_segments" value="false"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="5"/>

    <param name="max_group_distance"   value="0.1"/>
    <param name="distance_proportion"  value="0.1"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="0.2"/>
    <param name="max_merge_spread"     value="0.4"/>
    <param name="max_circle_radius"    value="20"/>
    <param name="radius_enlargement"   value="0.3"/>

    <param name="min_x_limit"   value="-40"/>
    <param name="max_x_limit"   value="40"/>
    <param name="min_y_limit"   value="-40"/>
    <param name="max_y_limit"   value="40"/>




    <param name="frame_id"             value="scan_frame"/>
  </node>

  <node name="obstacle_tracker" pkg="nodelet" type="nodelet" args="load obstacle_detector/ObstacleTracker nodelet_manager">
    <param name="active"                  value="false"/>

    <param name="loop_rate"               value="10.0"/>
    <param name="tracking_duration"       value="4.0"/>
    <param name="min_correspondence_cost" value="8.6"/>
    <param name="std_correspondence_dev"  value="0.15"/>
    <param name="process_variance"        value="0.1"/>  
    <param name="process_rate_variance"   value="0.1"/>  
    <param name="measurement_variance"    value="1.0"/>  

    <param name="frame_id"                value="scan_frame"/>

    <remap from="tracked_obstacles" to="obstacles"/>
  </node>


  <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find obstacle_detector)/resources/obstacle_detector.rviz"/>

</launch>
<!-- -->
